### PR TITLE
Make product page reviews responsive and apply design updates

### LINKED ISSUE #692: Make product page reviews section responsive
We have a reviews section on the product page but it's not optimised for mobile. There are also some design updates that we'd like to apply.

### PR DESCRIPTION
**Pull request by dhmacs**
closes #692 

## QA

1. Open Vercel preview
2. Visit a product with reviews
3. Verify that now the product reviews section is responsive

### CODE REVIEW

**Comment by jordycosta** [comment]
QA 🟢 

The product page reviews is responsive and design updates look good 👍 

### Testing Procedure:

1. Navigate to a product page
2. Scroll down to the reviews section
3. Modify your viewport (mobile, orientation, window size)
4. Ensure that the section stays responsive

### Screen Captures:

reviews
more_reviews

File: frontend/templates/product/sections/ReviewsSection/index.tsx [hunk 1]
```diff
@@ -220,7 +243,30 @@ type RatingCount = {
...
 function formatReviewDate(timeSeconds: number) {
-   return new Intl.DateTimeFormat('en-US', { dateStyle: 'long' }).format(
-      new Date(timeSeconds * 1000)
-   );
+   const formatter = new Intl.RelativeTimeFormat('en-US', { numeric: 'auto' });
+   const date = new Date(timeSeconds * 1000);
+   let duration = (date.getTime() - new Date().getTime()) / 1000;
+
+   for (let i = 0; i <= DIVISIONS.length; i++) {
+      const division = DIVISIONS[i];
```
Comment (masonmcelvain) [inline frontend/templates/product/sections/ReviewsSection/index.tsx:257]: Not worth blocking on, but I think we prefer `for (const division of DIVISIONS) { ... }` for readability if it's possible here

File: frontend/templates/product/sections/ReviewsSection/index.tsx [hunk 2]
```diff
@@ -220,7 +249,30 @@ type RatingCount = {
...
+   amount: number;
+   name: Intl.RelativeTimeFormatUnit;
+};
+
+const DIVISIONS: Division[] = [
+   { amount: 60, name: 'seconds' },
+   { amount: 60, name: 'minutes' },
+   { amount: 24, name: 'hours' },
+   { amount: 7, name: 'days' },
+   { amount: 4.34524, name: 'weeks' },
```
Comment (masonmcelvain) [inline frontend/templates/product/sections/ReviewsSection/index.tsx:275]: Took me a second to figure out that this is the average number of weeks in a month! That was a fun puzzle 😂

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻

**Pull request merged**